Celeste.Mod.mm/Patches/Decal.cs

```diff
@@ -97,7 +98,7 @@ private class DecalImage : Component {
                 OnShake = v => { X += v.X; Y += v.Y; },
                 OnAttach = p => {
                     p.Add(new EntityRemovedListener(() => RemoveSelf()));
-                    (Scene as Level).Session.SetFlag($"Everest_Decal_texture:{Name}_X:{Position.X}_Y:{Position.Y}_WasAttached");
+                    CoreModule.Session.AttachedDecals.Add($"{Name}||{Position.X}||{Position.Y}");
```

Comment:
does this break existing sessions on everest version upgrade?

Reply:
How would it? Would be pretty alarming if adding properties to a module’s session broke existing sessions on updating the mod, don’t see how Everest’s mod session would function any different. Not like Everest’s session was being used for anything else prior to this anyway, plus I ran around some levels with and without attached decals and nothing exploded in my face so not sure what else I’m supposed to be looking for

Reply:
i meant that existing sessions' attached flags would be ignored and some decals would reappear:
- player loads a file before this change, loads a room with a persistent dash block (w/ attached decal) and breaks it
- player updates everest
- player reenters map, and sees that the block is broken but the decal is back
is this a concern?

Reply:
Old maps that used staticMover decals never had those decals removed in the first place - the main focus of this pr is rectifying that via the EntityRemovedListener - so we're not "breaking" any existing behaviour there. The decal will just remain present in the room until the session is reset and the block comes back, and then the decals will start using the attachment behaviour being added here.

I guess it'd be nice if we could account for this and apply these fixes after-the-fact, but needing to add this functionality to track attached decals should show that we don't have anything at our disposal to determine if these existing decals were ever attached to something, so I'm not really seeing a way forward on this. If you come up with something then I'd appreciate if you could put it together yourself, I touched on it in celestecord but not in much of a position to keep chipping away at this, the time for me to do that was 4 weeks ago when I opened the pr